### PR TITLE
AOI Scribe Notes: Fault Attacks

### PR DESCRIPTION
**Pull request by drrl**
Chapter 9 - Fault Attacks

### CODE REVIEW

File: bibliography/ch9_fault_attacks.bib [hunk 1]
```diff
@@ -0,0 +1,63 @@
+@CONFERENCE {bib:nohl2008,
```
Comment (MahlerTom) [inline bibliography/ch9_fault_attacks.bib]: Need to add these into the existing english.bib file.

File: chapters/05_chapter_4.tex [hunk 2]
```diff
@@ -0,0 +1 @@
+\chapter{Replace with Chapter 4 Name}
```
Comment (MahlerTom) [inline chapters/05_chapter_4.tex]: Why are you adding this? This is for other students to add, please remove all such files. also from the main

File: main.tex [hunk 3]
```diff
@@ -63,8 +82,21 @@
...
+
+\input{chapters/09_chapter_8.tex}
+
+\input{chapters/10_chapter_9.tex}
+
 \clearpage
 \bibliographystyle{unsrt}
-\bibliography{bibliography/english}
+%\addbibresource{bibliography/english}
+\bibliography{bibliography/ch9_fault_attacks}
```
Comment (MahlerTom) [inline main.tex]: This overwrites the current bib. Please add your cites to the current file.

File: chapters/10_chapter_9.tex [hunk 4]
```diff
@@ -0,0 +1,329 @@
+\chapter{Fault Attacks} \label{c9_ninthchapter:cha}
```
Comment (MahlerTom) [inline chapters/10_chapter_9.tex:1]: Please use the convention chap:c9_.... for label

File: chapters/10_chapter_9.tex [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+\author{
+    \parbox{\linewidth}{\centering
+    Dorel Yaffe, Dan Arad, Iliya Fayans \\
+    \vspace{7pt}\texttt{\small{\{yaffed,arda,fayansi\}@post.bgu.ac.il}}
+    }
+}
+\date{\footnotesize{\textbf{2019-05-27}}}
+
+\maketitle
+
```
Comment (MahlerTom) [inline chapters/10_chapter_9.tex]: This is already done in the main. If you want to add your name, add it in the main. Don't touch the title or date.

File: chapters/10_chapter_9.tex [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+% Swap the definition of \abs* and \norm*, so that \abs
+% and \norm resizes the size of the brackets, and the 
+% starred version does not.
+\makeatletter
+\let\oldabs\abs
+\def\abs{\@ifstar{\oldabs}{\oldabs*}}
+%
+\let\oldnorm\norm
+\def\norm{\@ifstar{\oldnorm}{\oldnorm*}}
+\makeatother
```
Comment (MahlerTom) [inline chapters/10_chapter_9.tex]: Why is all this required?

File: chapters/10_chapter_9.tex [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+In fault attacks we are being \emph{active} -- we give additional inputs beside the main input such as:
+\begin{enumerate}
+	\item Fuzzing (garbage or bad input)
+	\item Radiation
+	\item Heat
+	\item Vibration
+	\item Power spikes etc.
+\end{enumerate}
+
+This way, we receive other (usually erronous) outputs which might give us additional information about the computation and/or the secret. This process is described in figure \ref{fig:fault_attacks_schematic}.
```
Comment (MahlerTom) [inline chapters/10_chapter_9.tex]: Please use cleveref \cref or \Cref for this.

File: chapters/10_chapter_9.tex [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+
+In an interesting research paper \cite{bib:appel} a type-confusion attack on the Java virtual-machine was demonstrated: at first, the entire memory was filled with small arrays (say of size one). The Java VM is type-safe, so it is normally impossible to access one of the memory regions using a pointer to a different region. To inject a type-confusion fault the researchers used a 50W lightbulb to heat the memory chip of the device enough to flip some of the bits (for illustration see Fig. \ref{fig:memory_lightbulb}). As a result, a small portion of the data-structures describing the arrays in memory now held wrong values (e.g. changed their value from $size=1$ to $size=20$). At this point, some affected data-structure \emph{contains} a header of a different data-structure, to which the attackers now have read and write access. Changing the header of the second data structure to an arbitrary value gave the attackers access to the entirety of the device's memory.
```
Comment (MahlerTom) [inline chapters/10_chapter_9.tex]: No need for the "bib" in each cite

File: chapters/10_chapter_9.tex [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+\begin{figure}[!ht]
+    \centering
+	\includegraphics[width=0.7\linewidth]{images/ch9/bulb.png}
+	\caption{A lightbulb flipping memory bits filled with safe Java structures.}
+	\label{fig:memory_lightbulb}
+\end{figure}
+
+\subsubsection{Optical, Electromagnetic}
```
Comment (MahlerTom) [inline chapters/10_chapter_9.tex]: Please label each section, subsection, etc. Use the convention {sec:...} or {subsec:...}, etc.

File: chapters/10_chapter_9.tex [hunk 10]
```diff
@@ -0,0 +1,329 @@
...
+
+\subsection{Rowhammer variations}
+\subsubsection{Flip Feng-Shui}
+\paragraph{Page de-duplication:} On modern systems, typically much memory is shared among many processes running on the system. This is even more true of virtualized environments where the guest and the host, for example, could run the same OS. A common optimization is for the system to detect and de-duplicate pages containing the same data, thus freeing up memory.
+\paragraph{Rowhammer + page de-duplication:}
+In a paper \cite{bib:ffs} researchers from VUA demostrated how they can utilize page-deduplication in a virtualized environment to weaken cryptographic keys, resulting in unauthorized access via OpenSSH, and breach of trust via forging GPG signatures. The attack relies on the fact that the the attacker can \emph{read} a de-duplicated page as much as they want. The attacker has to wait (or arrange) for a page containing sensitive information to be de-duped, then hammer on it until a bit in the key flips, making it much easier to factor.
+\begin{figure}[H]
```
Comment (MahlerTom) [inline chapters/10_chapter_9.tex]: Please use !ht in all figures

**Review by MahlerTom — CHANGES_REQUESTED** [review]
Overall excellent work. Note the changes and please apply them in all other places they appear. Also please don't submit any aux files.

File: chapters/10_chapter_9.tex (hunk 4 above)
Comment (dan1994) [reply chapters/10_chapter_9.tex:1]: From what I see in other chapters, this matches the convention.
Could you write the full convention?
Also I would recommend fixing this across the other chapters...

File: bibliography/ch9_fault_attacks.bib (hunk 1 above)
Comment (dan1994) [reply bibliography/ch9_fault_attacks.bib]: Done

File: chapters/05_chapter_4.tex (hunk 2 above)
Comment (dan1994) [reply chapters/05_chapter_4.tex]: Done

File: main.tex (hunk 3 above)
Comment (dan1994) [reply main.tex]: Reverted

File: chapters/10_chapter_9.tex (hunk 5 above)
Comment (dan1994) [reply chapters/10_chapter_9.tex]: Removed

File: chapters/10_chapter_9.tex (hunk 6 above)
Comment (dan1994) [reply chapters/10_chapter_9.tex]: Deleted

File: chapters/10_chapter_9.tex (hunk 7 above)
Comment (dan1994) [reply chapters/10_chapter_9.tex]: Changed to \cref.
Added usepackage in main.tex

File: chapters/10_chapter_9.tex (hunk 8 above)
Comment (dan1994) [reply chapters/10_chapter_9.tex]: Removed

File: chapters/10_chapter_9.tex (hunk 9 above)
Comment (dan1994) [reply chapters/10_chapter_9.tex]: Done

File: chapters/10_chapter_9.tex (hunk 10 above)
Comment (dan1994) [reply chapters/10_chapter_9.tex]: Done

**Pull request merged**